pkg/machineproviders/providers/openshift/machine/v1beta1/providerconfig/providerconfig.go

```diff
@@ -203,6 +207,11 @@ func getPlatformTypeFromMachineTemplate(tmpl machinev1.OpenShiftMachineV1Beta1Ma
 // getPlatformTypeFromProviderSpec determines machine platform from the providerSpec.
 // The providerSpec object's kind field is unmarshalled and the platform type is inferred from it.
 func getPlatformTypeFromProviderSpec(providerSpec machinev1beta1.ProviderSpec) (configv1.PlatformType, error) {
+	// providerSpec.Value might be nil, return early with error if so
+	if providerSpec.Value == nil {
+		return "", fmt.Errorf("%w: can't determine platform type, provider spec is empty", errProviderSpecIsEmpty)
```

Comment:
Nit, please don't abbreviate in error messages
```suggestion
		return "", fmt.Errorf("%w: cannot determine platform type, provider spec is empty", errProviderSpecIsEmpty)
```